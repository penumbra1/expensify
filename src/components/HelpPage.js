import React, { Fragment } from "react";

const HelpPage = props => (
  <Fragment>
    <div>Help</div>
    <button onClick={props.history.goBack}>Go back</button>
  </Fragment>
);

export default HelpPage;
