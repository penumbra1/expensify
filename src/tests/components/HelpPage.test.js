import React from "react";
import { shallow } from "enzyme";
import HelpPage from "../../components/HelpPage";

test("should render the help page correctly", () => {
  const history = { goBack: () => {} };
  const wrapper = shallow(<HelpPage history={history} />);

  expect(wrapper).toMatchSnapshot();
});

test("should go back in history on button click", () => {
  const history = { goBack: jest.fn() };
  const wrapper = shallow(<HelpPage history={history} />);

  wrapper.find("button").simulate("click");

  expect(history.goBack).toHaveBeenCalled();
});
