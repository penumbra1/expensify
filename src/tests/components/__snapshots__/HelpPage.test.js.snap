// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the help page correctly 1`] = `
<Fragment>
  <div>
    Help
  </div>
  <button
    onClick={[Function]}
  >
    Go back
  </button>
</Fragment>
`;
